### PR TITLE
Use `sass::sass_bundle()`

### LINKED ISSUE #105: Provide a way to generate slimmed down builds
i.e., provide a way to strip out rules

### PR DESCRIPTION
**Pull request by schloerke**
Fixes #105 

Can slim down by calling

```r
theme %>% sass_bundle_remove(NAME)
```

-----------------

## Pull Request

Before you submit a pull request, please ensure you've completed the following checklist

- [ ] Ensure there is an already open and relevant [GitHub issue](https://github.com/rstudio/bootstraplib/issues/new) describing the problem in detail and you've already received some indication from the maintainers that they are welcome to a contribution to fix the problem. This helps us to prevent wasting anyone's time. 

- [x] Add unit tests in the tests/testthat directory.

- [x] This project uses [roxygen2 for documentation](http://r-pkgs.had.co.nz/man.html). If you've made changes to documentation, run `devtools::document()`.

- [x] Run `devtools::check()` (or, equivalently, click on Build->Check Package in the RStudio IDE) to make sure your change did not add any messages, warnings, or errors.
    * Note there is a decent chance that some tests were already failing before your changes. Just make sure you haven't introduced any new ones.
    
- [x] Ensure your code changes follow the style outlined in http://r-pkgs.had.co.nz/style.html

- [ ] Add an entry to NEWS.md concisely describing what you changed.

-----------------------

To be validated after PR is merged: https://github.com/rstudio/sass/pull/54

### CODE REVIEW

File: R/files.R
Comment (cpsievert) [inline R/files.R:16]:
```suggestion
    ret <- lapply(ret, sass::sass_removable)
```

File: R/bs-theme.R
Comment (cpsievert) [inline R/bs-theme.R:145]:
```suggestion
    !!paste0(bootswatch, "@", version) := bootswatch_layer(bootswatch, version)
```

File: R/bs-theme-layers.R
Comment (cpsievert) [inline R/bs-theme-layers.R:125]:
```suggestion
  is_layer <- vapply(rlang::list2(...), is_sass_layer, logical(1))
```

File: R/bootswatch.R
Comment (cpsievert) [inline R/bootswatch.R]:
```suggestion
  matching_pos <- grep(paste0("^", name, "@(\\d)$"), layer_names)
```

File: R/bootswatch.R
Comment (cpsievert) [inline R/bootswatch.R:34]:
```suggestion
  if (length(matching_pos) > 1) stop("Found multiple ", name, " versions.")
```

File: R/bootswatch.R
Comment (cpsievert) [inline R/bootswatch.R:35]:
```suggestion
  sub(paste0(name, "@"), "", layer_names[matching_pos])
```

File: NAMESPACE
Comment (cpsievert) [inline NAMESPACE:7]: Maybe this should be `bs_bundle()` to complement the new `sass_bundle()`

File: R/bs-global.R
Comment (cpsievert) [inline R/bs-global.R:88]: Make sure to update the global API to reflect layer -> bundle changes

File: R/bs-theme-layers.R
Comment (cpsievert) [inline R/bs-theme-layers.R:122]: Add a `@describeIn` explaining the `...` and remove the (trailing) explanation from `bs_add_variables()`

File: R/bs-theme-layers.R
Comment (cpsievert) [inline R/bs-theme-layers.R:129]: Remove 126-129

File: R/bs-theme.R
Comment (cpsievert) [inline R/bs-theme.R:229]: Need to distinguish between name version and file version

Comment (cpsievert) [reply R/bs-theme.R:229]: Example: file version is 4 and name version is `4+3`

File: R/files.R
Comment (cpsievert) [inline R/files.R:79]:
```suggestion
  sass_bundle(
    !!bs_sass_bundle_version(name, version) := core, 
    sass_bundle(!!!rule_layers)
  )
```

File: R/bs-theme.R
Comment (cpsievert) [inline R/bs-theme.R:340]: Won't need the `list()` wrappers after sass changes

File: R/bs-theme.R
Comment (cpsievert) [inline R/bs-theme.R:290]:
```suggestion
      !!bs_sass_bundle_version("bootstrap", version, subname = "glyphicon") := glyphicon
```

File: R/bs-theme.R
Comment (cpsievert) [inline R/bs-theme.R:345]: No need for `list()`

File: R/bs-theme.R
Comment (cpsievert) [inline R/bs-theme.R:362]: No need for `list()`

File: R/bs-theme.R
Comment (cpsievert) [inline R/bs-theme.R:371]: No need for `list()`

File: R/bs-theme-layers.R
Comment (cpsievert) [inline R/bs-theme-layers.R:9]:
```suggestion
#'  * `bs_add_variables()`: Should be named Sass variables or values that can be passed in directly to the `defaults` argument of a [sass::sass_layer()].
```

File: R/bs-theme.R
Comment (cpsievert) [inline R/bs-theme.R:293]: I went back-and-forth on whether to even name this bundle at all (since it just let's the CSS rules know where to find font files), but I suppose it might be useful to avoid the overhead of copying over files

```suggestion
      !!bs_sass_bundle_version("bootstrap", version, subname = "glyphicon_fonts") := glyphicon
```

File: R/files.R
Comment (cpsievert) [inline R/files.R:88]: I was thinking it might helpful to use the `subname` to tag a set `file`s, so it seems like it'd make sense for `subname` to appear before `file` (i.e., the naming goes broad -> narrow)?

File: tests/testthat/test-zzzz-bs-sass.R
Comment (cpsievert) [inline tests/testthat/test-zzzz-bs-sass.R:4]: What's the rationale for this?

**Review by cpsievert — APPROVED** [review]
LGTM pending suggestions! 🎉

File: DESCRIPTION
Comment (cpsievert) [inline DESCRIPTION:61]: Let's make sure to remove this after merging the sass PR

```suggestion
    rstudio/sass,
```

File: tests/testthat/test-zzzz-bs-sass.R
Comment (schloerke) [reply tests/testthat/test-zzzz-bs-sass.R:4]: These tests take forever and by the time CRAN gets them, it would be too late. At least we can do these tests on GHA

File: R/files.R
Comment (schloerke) [reply R/files.R:88]: Fair!

File: R/bs-theme-preview.R
Comment (cpsievert) [inline R/bs-theme-preview.R:398]: Why is this needed?

Comment (schloerke) [reply R/bs-theme-preview.R:398]: When the newly updated Sass compiles it, it was re-ordered.

**Comment by schloerke** [comment]
Going to merge. Ubuntu has R installation errors. Not checking errors. Ubuntu 20 passed.

**Pull request merged**
